tests/unit/customizations/test_scalarparse.py

```diff
@@ -30,10 +36,89 @@ def test_identity(self):
         self.assertEqual(scalarparse.identity(10), 10)
 
     def test_scalar_parsers_set(self):
-        session = mock.Mock()
+        session = Mock()
+        session.get_scoped_config.return_value = {'cli_timestamp_format':
+                                                  'none'}
         scalarparse.add_scalar_parsers(session)
         session.get_component.assert_called_with('response_parser_factory')
         factory = session.get_component.return_value
+        expected = [call(blob_parser=scalarparse.identity),
+                    call(timestamp_parser=scalarparse.identity)]
+        self.assertEqual(factory.set_parser_defaults.mock_calls,
+                         expected)
+
+    def test_choose_none_timestamp_formatter(self):
+        session = Mock(spec=Session)
+        session.get_scoped_config.return_value = {'cli_timestamp_format':
+                                                  'none'}
+        factory = session.get_component.return_value
+        scalarparse.add_scalar_parsers(session)
+        factory.set_parser_defaults.assert_called_with(
+            timestamp_parser=scalarparse.identity)
+
+    def test_choose_iso_timestamp_formatter(self):
+        session = Mock(spec=Session)
+        session.get_scoped_config.return_value = {'cli_timestamp_format':
+                                                  'iso8601'}
+        factory = session.get_component.return_value
+        scalarparse.add_scalar_parsers(session)
+        factory.set_parser_defaults.assert_called_with(
+            timestamp_parser=scalarparse.iso_format)
+
+    def test_choose_invalid_timestamp_formatter(self):
+        session = Mock(spec=Session)
+        session.get_scoped_config.return_value = {'cli_timestamp_format':
+                                                  'foobar'}
+        session.get_component.return_value
+        with self.assertRaises(ValueError):
+            scalarparse.add_scalar_parsers(session)
+
+    def test_choose_timestamp_parser_profile_not_found(self):
+        session = Mock(spec=Session)
+        session.get_scoped_config.side_effect = ProfileNotFound(profile='foo')
+        factory = session.get_component.return_value
+        scalarparse.add_scalar_parsers(session)
         factory.set_parser_defaults.assert_called_with(
-            blob_parser=scalarparse.identity,
             timestamp_parser=scalarparse.identity)
+
+
+class TestCLITimestampParser(unittest.TestCase):
```

Comment:
So this looks more like a functional test. Since this is a super set of what your other functional test you added tests, I would suggest just to remove ``TestParsers`` and add ``TestCLITimestampParser`` to/from ``tests/functional/test_timeformat.py``.

Also it would be great if there was a functional test for both ``none`` being set and having it not configured at all for completeness.